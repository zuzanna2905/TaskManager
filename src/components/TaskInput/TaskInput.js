import React from 'react';
import queryString from 'query-string';

class TaskInput extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      input: ''
    }
  }

  handleFormChange = (event) => {
    const {name, value} = event.target; 
    this.setState({[name]: value})
  }

  onSubmitTask = () => {
    if(this.state.input){
      fetch('http://localhost:3001/tasks?' + queryString.stringify({
          userid: this.props.userid, 
          task: this.state.input
        }), {
          method: 'post',
          headers: {'Content-Type' : 'application/json'}
      })
      .then(response => { 
        response.json();
        this.setState({input: ''}, () => this.props.loadData());
        })
    }
  }

  onKeyPress = (event) => {
    if (event.key === 'Enter') {
      this.onSubmitTask();
    }
  }

  render(){
    return (
      <div>
          <div className="flex center w-80">
              <input 
                id="task" 
                onChange={this.handleFormChange} 
                onKeyPress={this.onKeyPress}
                name='input'
                value={this.state.input}
                className="ma2 bg-white-60 input-reset ba b--black-20 br4 pa2 db w-80" 
                placeholder='Task...' 
                type="text" 
                aria-describedby="task-desc"
              />
              <button 
                id='submittask'
                onClick={() => this.onSubmitTask}
                className="black br4 bg-white-40 mh2 mv1 f5 ph2 w-20"
              > Add </button>
          </div>
      </div>
    )
  }
}

export default TaskInput
