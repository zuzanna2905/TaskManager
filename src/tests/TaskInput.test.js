import {shallow} from 'enzyme';
import React from 'react';
import TaskInput from '../components/TaskInput/TaskInput';

it('renders TaskInput without crashing', () => {
    let wrapper = shallow(<TaskInput />);
    expect(wrapper).toMatchSnapshot();
});

it('simulate TaskInput', () => {    
    const expectedState = {
        input: 'kwiatki'
    }
    let wrapper2 = shallow(<TaskInput />);
    wrapper2.find('[id="task"]').simulate('change', {target: {value: 'kwiatki', name:'input'}});
    expect(wrapper2.state().input).toEqual(expectedState.input);
    wrapper2.find('[id="submittask"]').simulate('click');  
})