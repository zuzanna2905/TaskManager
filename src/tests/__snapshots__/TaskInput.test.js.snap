// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TaskInput without crashing 1`] = `
<div>
  <div
    className="flex center w-80"
  >
    <input
      aria-describedby="task-desc"
      className="ma2 bg-white-60 input-reset ba b--black-20 br4 pa2 db w-80"
      id="task"
      name="input"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Task..."
      type="text"
      value=""
    />
    <button
      className="black br4 bg-white-40 mh2 mv1 f5 ph2 w-20"
      id="submittask"
      onClick={[Function]}
    >
       Add 
    </button>
  </div>
</div>
`;
